pandas/core/window/rolling.py

```diff
@@ -1871,7 +1875,7 @@ def _validate(self):
         # we allow rolling on a datetimelike index
         if (
             self.obj.empty
-            or isinstance(self._on, (DatetimeIndex, TimedeltaIndex, PeriodIndex))
+            or (isinstance(self._on, PeriodIndex) or self._on.dtype.kind in "Mm")
```

Comment:
id err on the side of being more explicit about the supported dtypes being ArrowDtype